### ISSUE TITLE
Constructor for an interpolant which is the derivative of another

### ISSUE THREAD

**Issue by danmackinlay** [JuliaMath/Interpolations.jl#320]
Currently we have `hessian` and `gradient` to evaluate derivatives of `Interpolation` objects. However, sometimes we wish to simply construct a whole *new* interpolation which is the derivative of another and use this.
(also, more generally, we might want to construct interpolations  which are integrals of other interpolations, but there are a couple of other details there). 

It looks like this should be mathematically not too complex, but I got lost trying to understand how this would get the correct order and boundary conditions.

**Comment by timholy** [comment]
Yes, that's a reasonable thing to want. Again, a PR would be welcome.
